### PR TITLE
Add DynamicBmp

### PR DESCRIPTION
**Pull request by rfuest**
Hi! Thank you for helping out with Embedded Graphics development! Please:

- [ ] Check that you've added passing tests and documentation
- [ ] Add a simulator example(s) where applicable
- [ ] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) and appropriate crate (`embedded-graphics`, `simulator`, `tinytga`, `tinybmp`) if your changes affect the **public API**
- [ ] Run `rustfmt` on the project
- [ ] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

This PR adds `DynamicBmp` and reverts some previous changes by splitting `Bmp` into `Bmp` and `BmpRaw`. It also contains an extended parser for the BMP header which is required to distinguish between `Rgb555` and `Rgb565` images.

Because the code depends on #450 and is also missing some docs the PR is currently marked as a draft. I'll finish it after the other PR is merged.

### CODE REVIEW

File: tinybmp/src/raw_bmp.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+use embedded_graphics::{prelude::*, primitives::Rectangle};
+
+use crate::{
+    header::{Bpp, Header},
+    pixels::Pixels,
+    raw_pixels::RawPixels,
+    ParseError,
+};
+
+/// A BMP-format bitmap
```
Comment (jamwaffles) [inline tinybmp/src/raw_bmp.rs:10]:
```suggestion
/// A BMP-format bitmap.
```

File: tinybmp/src/raw_bmp.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct RawBmp<'a> {
+    /// Image header
```
Comment (jamwaffles) [inline tinybmp/src/raw_bmp.rs:13]:
```suggestion
    /// Image header.
```

File: tinybmp/src/raw_bmp.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    header: Header,
+
+    /// Image data
```
Comment (jamwaffles) [inline tinybmp/src/raw_bmp.rs:16]:
```suggestion
    /// Image data.
```

**Review by jamwaffles — CHANGES_REQUESTED** [review]
I left some tiny nits so I can still feel relevant. Otherwise this looks good to go.

**Review by jamwaffles — APPROVED** [review]
shipit

**Comment by rfuest** [comment]
Thanks for the review!

**Comment by jamwaffles** [comment]
Any time :)

**Pull request merged**